src/utils/cachingAssets.ts

```diff
@@ -0,0 +1,43 @@
+import { ApolloClient, NormalizedCacheObject, gql } from '@apollo/client'
+
+type ReadUrlFromCacheArg = {
+  fileType: 'avatar' | 'cover'
+  channelId: string | null
+  client: ApolloClient<NormalizedCacheObject>
+}
+
+type WriteUrlInCacheArg = {
+  url: string | null
+} & ReadUrlFromCacheArg
+
+const cachedCoverUrlFragment = gql`
+  fragment CoverUrlField on Channel {
+    cachedCoverUrl
+  }
+`
+
+const cachedAvatarUrlFragment = gql`
+  fragment AvatarUrlField on Channel {
+    cachedAvatarUrl
+  }
+`
+
+export const writeUrlInCache = ({ url, fileType, channelId, client }: WriteUrlInCacheArg) => {
+  const field = fileType === 'avatar' ? 'cachedAvatarUrl' : 'cachedCoverUrl'
+  client.writeFragment({
+    id: `Channel:${channelId}`,
+    fragment: fileType === 'avatar' ? cachedAvatarUrlFragment : cachedCoverUrlFragment,
+    data: {
+      [field]: url,
+    },
+  })
+}
+
+export const readUrlFromCache: (arg: ReadUrlFromCacheArg) => string = ({ fileType, channelId, client }) => {
```

Comment:
And once we move away from the custom fields, I don't think we need the read function at all right?

Reply:
that's correct